src/nxt_http_static.c

```diff
@@ -1,6 +1,7 @@
 
 /*
  * Copyright (C) NGINX, Inc.
+ * Copyright 2021-2024, Alejandro Colomar <alx@kernel.org>
```

Comment:
Respectfully, I'd like to block merging until I can figure out F5's processes around managing copyright and attribution in F/OSS projects. I'll do my best to track that down as quickly as possible.

Reply:
Sure; thanks!  Mind that 2021-2024, comes from the following.

In the period between 2021 and 2022 I contributed code to Unit before I worked for F5, as an external voluntary contributor, as per the agreement between VBart, Max, and me.

Reply:
Thank you! I'm pretty new here, so trying to make sure I do the right thing by the project and its contributors.

Your external -> internal -> external arc makes me wonder if we'll need to do something tedious like "2021-2022, 2024", omit the year altogether, move credits to a `NOTICE` file, or something entirely different. It's been a while since I've worked with Apache-2.0 licensed software. 😅

Reply:
@alejandro-colomar Chatted with F5 Legal and their preferred direction would be to update the years to exclude the time when you were working for F5. Less preferred but fine on our end would be to exclude the years altogether.

Reply:
Sure, a bit ugly but more correct.  I'll update that.

Reply:
```
$ git range-diff master gh/mtypes mtypes 
 1:  ec39c37a =  1:  ec39c37a HTTP: static: Use shr instead of ctx->share
 2:  cc01caef !  2:  75257300 HTTP: static: Use a common variable in both index and non-index code
    @@ src/nxt_http_static.c
      
      /*
       * Copyright (C) NGINX, Inc.
    -+ * Copyright 2021-2024, Alejandro Colomar <alx@kernel.org>
    ++ * Copyright 2021-2022, 2024, Alejandro Colomar <alx@kernel.org>
       */
      
      #include <nxt_router.h>
 3:  a2fc2fab =  3:  454cc926 HTTP: static: Split conditional
 4:  ce01dfc4 =  4:  f5f9caa6 HTTP: static: Use path instead of shr for extracting the extension
 5:  57899564 =  5:  d4c1bb1c HTTP: static: Don't include the terminating null byte in exten
 6:  104433e5 =  6:  ee1912d8 HTTP: static: Deduplicate code
 7:  ad56ec75 =  7:  e2f67946 HTTP: static: Remove dead code
 8:  9a8e6d92 =  8:  e25c80ea HTTP: static: Terminate string with a null byte
 9:  3d567775 =  9:  4048c67e HTTP: static: Honor (MIME) "types" in "index"
10:  7feceaf3 = 10:  47c5f1e9 HTTP: static: Deduplicate code
```